src/devno.rs

```diff
@@ -9,6 +9,30 @@ use crate::err::Result;
 /// Device number
 pub struct BlkidDevno(libc::dev_t);
 
+#[cfg(target_os = "linux")]
+pub mod ty {
+    /// Type alias for device major number
+    #[allow(non_camel_case_types)]
+    pub type maj_t = libc::c_uint;
+
+    /// Type alias for device minor number
+    #[allow(non_camel_case_types)]
+    pub type min_t = libc::c_uint;
+}
+
+#[cfg(all(unix, not(target_os = "linux"), not(target_os = "android")))]
```

Comment:
This should cover macOS and all BSD systems. I also specifically excluded android because the definitions in libc are really messed up.

```
pub {const} fn makedev(ma: c_uint, mi: c_uint) -> crate::dev_t {
    let ma = ma as crate::dev_t;
    let mi = mi as crate::dev_t;
    ((ma & 0xfff) << 8) | (mi & 0xff) | ((mi & 0xfff00) << 12)
}

pub {const} fn major(dev: crate::dev_t) -> c_int {
    ((dev >> 8) & 0xfff) as c_int
}

pub {const} fn minor(dev: crate::dev_t) -> c_int {
    ((dev & 0xff) | ((dev >> 12) & 0xfff00)) as c_int
}
```

The min/maj type is not even consistent there, so I don't want to complicate it.

Reply:
I think this is okay. I'm not even sure if there's a use case for libblkid on Android anyway. If there is, we can revisit this in the future.